pkg/reconciler/pipelinerun/resources/apply.go

```diff
@@ -237,10 +242,12 @@ func ApplyTaskResultsToPipelineResults(
 	results []v1beta1.PipelineResult,
 	taskRunResults map[string][]v1beta1.TaskRunResult,
 	customTaskResults map[string][]v1alpha1.RunResult) []v1beta1.PipelineRunResult {
-
+	// TODO(#4723): Validate array and object results.
```

Comment:
I will add more details in the comment, we need to validate the pipelineresults reference is 

1. array indexing reference: not out of bound 
2. object element reference: the key is in the dict

Reply:
ok, how hard would it be to add the validation in this PR? I wouldn't want this to be released without the validation

Reply:
I think adding a TODO for validation is to to avoid having those XL PRs, but I agree that we shouldn't hurry when the validations are not there. I will move the issue to next release if the validations can't be merged in time

Reply:
I believe most of TEP 0075, 0076 are finished. The only two features with no validations are: pipeline results for object types and array indexing for both pipelines and tasks. I can remove this two features from v0.38 release.